syntax = "proto3";

option java_multiple_files = true;
option java_package = "grpc.lamp";
option java_outer_classname = "LampProto";

package lamp;

enum LampType {
  ON_OFF = 0;
  BRIGHTNESS = 1;
}

message LampRequest {
  string id = 1;
  int32 value = 2;
}

message LampResponse {
  string message = 1;
}

message LampInfoRequest {
  string id = 1;
}

message LampInfoResponse {
  string id = 1;
  LampType type = 2;
  int32 value = 3;
}

service LampService {
  rpc SetLamp (LampRequest) returns (LampResponse) {}
  rpc GetInfo (LampInfoRequest) returns (LampInfoResponse) {}
}